### PR TITLE
Connect LedgerAPI party and package management with KVUtils

### PR DESCRIPTION
**Pull request by mziolekda**
- Provide glue code that connects Ledger API services for party and package management with participant state interfaces and ultimately with the KVUtils-based ledgers.

### CODE REVIEW

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/LedgerApiTestTool.scala [hunk 1]
```diff
@@ -138,7 +143,39 @@ object LedgerApiTestTool {
             commonConfig.withDarFile(resourceAsFile(semanticTestsResource))
       }
     )
-    Map(semanticTestsRunner)
+    val packageManagementServiceIT = lazyInit(
```
Comment (gerolf-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/LedgerApiTestTool.scala]: I'd rather we don't add tests to the "default set of tests" until we have the tests also running against sandbox via the test tool. This is even more the case for tests that we semi-instantiate but don't make available anyway, like the `PartyManagementServiceIT` test.

Comment (mziolekda) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/LedgerApiTestTool.scala]: Moved to optional part.

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/LedgerApiTestTool.scala [hunk 2]
```diff
@@ -246,12 +217,43 @@ object LedgerApiTestTool {
...
       transactionBackpressureIT,
       divulgenceIT,
       commandTransactionChecksHighLevelIT,
-      commandTransactionChecksLowLevelIT
+      commandTransactionChecksLowLevelIT,
+      packageManagementServiceIT
+      // TODO: Enable when KVUtils participant is backed by a postgres index database
+      // Present implementation is susceptible to race conditions between allocateParty
+      // and listKnownParties
+      //partyManagementServiceIT
```
Comment (gerolf-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/LedgerApiTestTool.scala]: Please un-comment the test. It is optional, so users have to opt-in anyway.

Comment (mziolekda) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/LedgerApiTestTool.scala]: Done

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/SandboxServer.scala [hunk 3]
```diff
@@ -82,6 +83,9 @@ object SandboxServer {
 
 class SandboxServer(actorSystemName: String, config: => SandboxConfig) extends AutoCloseable {
 
+  // Name of this participant, ultimately pass this info in command-line
+  val participantId: ParticipantId = Ref.LedgerString.assertFromString("in-memory-participant")
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/SandboxServer.scala]: Please use `sandbox` as the participant id.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/SandboxIndexAndWriteService.scala [hunk 4]
```diff
@@ -385,7 +396,7 @@ abstract class LedgerBackedIndexService(
   override def getParticipantId(): Future[ParticipantId] =
     // In the case of the sandbox, there is only one participant node
     // TODO: Make the participant ID configurable
-    Future.successful(ParticipantId(ledger.ledgerId.unwrap))
+    Future.successful(Ref.LedgerString.assertFromString(ledger.ledgerId.unwrap))
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/SandboxIndexAndWriteService.scala]: We can return the participant ID here, no?

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/PostgresLedgerDao.scala [hunk 5]
```diff
@@ -963,6 +963,8 @@ private class PostgresLedgerDao(
     dbDispatcher.executeSql { implicit conn =>
       SQL_SELECT_PARTIES
         .as(PartyDataParser.*)
+        // TODO: isLocal should be based on equality of participantId reported in an
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/PostgresLedgerDao.scala]: Please open a github issue for this and reference the issue here in a code comment.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/SandboxIndexAndWriteService.scala (hunk 4 above)
Comment (mziolekda) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/SandboxIndexAndWriteService.scala]: indeed, thanks for spotting this; that was the very reason for this entire storm in the glass

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/index/StandaloneIndexServer.scala [hunk 6]
```diff
@@ -79,6 +80,9 @@ class StandaloneIndexServer(
     readService: ReadService,
     writeService: WriteService) {
 
+  // Name of this participant, ultimately pass this info in command-line
+  val participantId: ParticipantId = Ref.LedgerString.assertFromString("in-memory-participant")
```
[the remark this answers is not in the document]
Comment (mziolekda) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/index/StandaloneIndexServer.scala]: Done

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/SandboxServer.scala (hunk 3 above)
Comment (mziolekda) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/SandboxServer.scala]: Done

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/SandboxIndexAndWriteService.scala (hunk 4 above)
Comment (mziolekda) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/SandboxIndexAndWriteService.scala]: Done

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/PostgresLedgerDao.scala (hunk 5 above)
Comment (mziolekda) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/PostgresLedgerDao.scala]: Issue #2026

**Pull request merged**